### ISSUE TITLE
mso_schema_site not working - 

### ISSUE THREAD

**Issue by richwats** [CiscoDevNet/terraform-provider-mso#7]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.14.4
+ provider registry.terraform.io/ciscodevnet/mso v0.1.4

### MSO version
* MSO 3.0(2d)

### APIC version and APIC Platform for Site Level Resources

* CAPIC 5.0(2h)

### Affected Resource(s)

mso_schema_site

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

data "mso_tenant" "Production" {
  name = "Production"
  display_name = "Production"
}

data "mso_site" "AWS-SYD" {
  name  = "AWS-SYD"
}

resource "mso_schema" "tf-hybrid-cloud" {
  name          = "tf-hybrid-cloud"
  template_name = "tf-hc-prod"
  tenant_id     = data.mso_tenant.Production.id
}

resource "mso_schema_site" "AWS-SYD" {
  schema_id  = mso_schema.tf-hybrid-cloud.id
  site_id  = data.mso_site.AWS-SYD.id
  template_name  = mso_schema.tf-hybrid-cloud.template_name
}

```

### Debug Output

https://gist.github.com/richwats/33af7b3d485687916847314cea038438

### Expected Behavior

Association of AWS site to schema & template

### Actual Behavior

Request to associate schema & template with site fails 

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

* Existing Tenant & Site in MSO already

### References

mso/api/v1/schemas
..specifically the "sites" array field.  It looks like its rejecting the call to this API as the Regions field is mandatory but not set using this terraform function.  There are separate functions (mso_schema_site_vrf, mso_schema_site_vrf_region etc.) to set these but can't be used without "mso_schema_site" being used first.

**Comment by lhercot** [comment]
This issue is related to the backend API validations. We are working with the backend team to get this resolved.
In the meantime, you can push the site config using mso_rest resources with at least the region and CIDR config for the VRFs in that site.

**Comment by richwats** [comment]
Hi Lionel,

I’m trying to do this now, but not having much luck.     I have gone to Postman to test out the actual REST call and I find while I can do this with PUT request, I can’t do this with a PATCH request (following MSO’s PATCH methodology)

I’m not sure I can do PUT type requests easily in Terraform?  I’d need to get the full GET response first then, in this case, add the “sites” array to the bottom of the JSON before sending the full JSON body back…

Do you have any examples on how to do this?

Thank you,
Richard

**Comment by lhercot** [comment]
Hi Richard,

The example I have use PATCH API to achieve this. They might have issue when destroy is called if used in a separate plan.

Example using PATCH with AWS site: https://github.com/lhercot/terraform-mso-demo_aws/blob/master/main.tf
Example using PATCH with Azure site: https://github.com/lhercot/terraform-mso-demo_azure/blob/master/main.tf

**Comment by anvitha-jain** [comment]
This issue is resolved, and the changes are in #52. This will be soon be available in the next release.
